app/controllers/orders_controller.rb

```diff
@@ -21,35 +22,57 @@ def update
     render :show
   end
 
+  def index
+    @merchant = User.find(params[:user_id])
 
-  def destroy
-    @order.destroy
-  end
+    @all_items = @merchant.order_items
 
-  def index
-    @all_orders = Order.all
+    @pend = @all_items.where(status: "pending")
+    @paid = @all_items.where(status: "paid")
+    @ship = @all_items.where(status: "complete")
+
+    @pend_total = @all_items.where(status: "pending").pluck(:total_price)
+    @paid_total = @all_items.where(status: "paid").pluck(:total_price)
+    @ship_total = @all_items.where(status: "complete").pluck(:total_price)
   end
 
   def show
+    @order_items = current_order.order_items
+    @order = Order.find(session[:order_id])
+    @total = @order.subtotal
+  end
 
+  def buyer
+    @buyer =  Buyer.find(params[:id])
+    @merchant = User.find(params[:user_id])
+  end
+
+  # check to see if the correct user is logged in
+  # renamed the method since I needed the user_id, not the id
+  def correct_order
+    @user = User.find(params[:user_id])
+    redirect_to(root_url) unless current_user?(@user)
   end
 
   private
 
-  def order_params
-    params.require(:order).permit(:subtotal,
-                                  :buyer_name,
-                                  :buyer_email,
-                                  :buyer_zip,
-                                  :buyer_state,
-                                  :buyer_address,
-                                  :buyer_city,
-                                  :buyer_expcc,
-                                  :buyer_last4cc)
-  end
+    def order_params
+      params.require(:order).permit(:id)
+    end
 
-  def find_order
-    @order = Order.find(id: order_params[:id])
-  end
+    def find_order
+      @order = Order.find(id: order_params[:id])
+    end
+
+    def empty_cart?
+      if session[:order_id].nil?
+        render :empty
+      elsif session[:order_id].nil? == false
```

Comment:
Since this line is actually just checking the inverse of the `if` statement, you should just use an `else` here rather than adding another condition